lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3187,6 +3187,10 @@ private function eagerLoadCollections(array $collections, array $mapping): void
             foreach ($found as $targetValue) {
                 $sourceEntity = $targetProperty->getValue($targetValue);
 
+                if ($sourceEntity === null) {
+                    return;
```

Comment:
Exclude collection initialization if one of the collection elements has not completed full hydration